dev-packages/electron/electron-codecs-test.js

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env node
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+'use-strict'
+
+// @ts-check
+
+const yargs = require('yargs');
+
+const { platforms, libffmpegCodecs, libffmpegAbsolutePath } = require('./electron-ffmpeg-lib');
+
+const bad = new Set([
+    'h264',
+    'aac',
+]);
+
+async function main() {
+    const options = yargs
+        .option('absolutePath', {
+            alias: ['a'],
+            description: 'Absolute path to the ffmpeg shared library.',
+        })
+        .option('electronDist', {
+            alias: ['d'],
+            description: 'Electron distribution location.',
+        })
+        .option('platform', {
+            alias: ['p'],
+            description: 'Dictates where the library is located within the Electron distribution.',
+            choices: platforms,
+        })
+        .help().alias('h', 'help')
+        .exitProcess(false)
+        .argv;
+
+    if (options.help) {
+        return; // help is being displayed.
+    }
+    const libraryPath = options['absolutePath'] || libffmpegAbsolutePath({
+        electronDist: options['electronDist'],
+        platform: options['platform'],
+    });
+    const codecs = libffmpegCodecs(libraryPath);
+    const found = [];
+    for (const codec of codecs) {
+        if (bad.has(codec.name.toLowerCase())) {
+            found.push(codec);
+        }
+    }
+    if (found.length > 0) {
+        throw new Error(`${found.length} bad / ${codecs.length} found\n${
+            found.map(codec => `> ${codec.name} detected (${codec.longName})`).join('\n')}`);
+    }
+    console.info(`"${libraryPath}" does not contain proprietary codecs (${codecs.length} found).`);
```

Comment:
It's about logging only: I do not see this when I run `yarn`. It would be useful to indicate that this check has run and terminated successfully. (I am not sure if it is possible.)

This is the entire output until the `prepare` phase:
```
kittaakos@DESKTOP-VJRAMKC MINGW64 ~/dev/theia (mp/h264-native-2)
$ yarn
yarn install v1.10.1
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
warning monaco-languageclient@0.9.0: The engine "vscode" appears to be invalid.
warning vscode-base-languageclient@4.4.0: The engine "vscode" appears to be invalid.
info fsevents@1.2.4: The platform "win32" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning " > istanbul-instrumenter-loader@3.0.1" has unmet peer dependency "webpack@^2.0.0 || ^3.0.0 || ^4.0.0".
warning " > tslint-language-service@0.9.9" has incorrect peer dependency "typescript@>= 2.3.1 < 3".
warning "workspace-aggregator-f7d56b5f-602d-42a8-a391-3f6448546a61 > @theia/application-manager > font-awesome-webpack@0
.0.5-beta.2" has unmet peer dependency "font-awesome@>=4.3.0".
warning "workspace-aggregator-f7d56b5f-602d-42a8-a391-3f6448546a61 > @theia/debug > requestretry@3.1.0" has unmet peer d
ependency "request@2.*.*".
warning "workspace-aggregator-f7d56b5f-602d-42a8-a391-3f6448546a61 > @theia/application-manager > font-awesome-webpack >
 less-loader@2.2.3" has incorrect peer dependency "less@^2.3.1".
[5/5] Building fresh packages...
warning Your current version of Yarn is out of date. The latest version is "1.16.0", while you're on "1.10.1".
$ node scripts/skip-prepare || (yarn prepare:travis && yarn rebuild:clean && yarn build:clean && yarn prepare:hoisting)
yarn run v1.10.1
$ node scripts/prepare-travis
Done in 3.17s.
yarn run v1.10.1
$ rimraf .browser_modules
Done in 0.44s.
yarn run v1.10.1
$ run prepare
Parallel lerna execution is disabled on Windows. Falling back to sequential execution with the '--concurrency=1' flag.
Running lerna as: C:\Program Files\nodejs\node.exe C:\Users\kittaakos\dev\theia\dev-packages\ext-scripts\theia-run run p
repare --concurrency=1
lerna info version 2.11.0
@theia/ext-scripts: $ echo 'skip'
```

If I run `yarn --cwd ./dev-packages/electron/ postinstall` manually, I can see
```
"C:\Users\kittaakos\dev\theia\node_modules\electron\dist\ffmpeg.dll" does not contain proprietary codecs (16 found).
```

Reply:
The output is mostly useful when people manually run the test.

When part of the installation process, it will fail everything if the test doesn't pass.
But since yarn suppresses the output I don't know what to do.

Reply:
Are you fine as it is or do you think it should really explicitly appear?

Reply:
To me this would be a crucial part, we do this IP validation to see everything is/was fine.

 - I think we have to use `process.stdio.write` instead.
 - We need to append an `\n` to the line. Otherwise, it is not visible: https://github.com/yarnpkg/yarn/issues/5005
 - And perhaps we have to adjust the `stdio` for the `fork` options.

But I won't hold back this PR. If you think it is not important, we do not do it.

Also, please note, you have to find someone who will eventually approve the PR. I cannot do it to the C++ code.

Reply:
> To me this would be a crucial part, we do this IP validation to see everything is/was fine.

I agree, but Yarn really did not want display anything. I tried a lot of different things, but the solution I came up with is to write a to a log file and then display it on `postinstall` from the npm/yarn monorepo root.

I added timestamps to help identify if everything ran during the most recent build.

Reply:
OK. Let's leave it as is.